docs/sources/write/style-guide/write-for-developers/index.md

```diff
@@ -78,23 +78,110 @@ Additional tips:
+
+Spaces (` `) are a good default but note that some languages use alternative indentation.
+For example, Makefiles and Go source code uses tabs (`	`) for indentation.
+
+Use 2 spaces unless there is a clearly established alternative convention.
+For example, Python generally uses 4 spaces for indentation.
+
+Above all else, be consistent with other Grafana documentation.
+
+#### Introduce each code sample with a sentence or paragraph to establish its context
+
+End the introduction with a colon (`:`) if it immediately precedes the sample, or a period (`.`) if it doesn't.
+
+#### Use the `code` shortcode for tabbed examples
+
+If you have the same example in multiple languages, use the [`code` shortcode](https://grafana.com/docs/writers-toolkit/write/shortcodes/#code).
+The website presents the snippets in tabs and remembers the users preferred choice.
+
+### Format partial code examples
+
+Partial code examples are shorter and focus the reader on a specific area of the code.
+However, they require the user to integrate the partial code example with their existing configuration or other examples.
+
+In addition to the general guidance for formatting code examples, when working with partial code examples, use the following guidelines.
+
+#### Wherever possible, ensure the partial example is copy-pasteable
+
+In JSON or YAML, the partial example should be valid on its own.
+
+In programming language code examples, if the example can't be compiled on its own, make sure the snippet is a recognizable unit of a source file.
+
+Do not use an ellipsis (`…`) or three periods (`...`) to omit information.
+It breaks the readers ability to copy-paste the example and provides no additional context for the omission.
+Instead use the preceding sentences or valid code comments to clearly explain the scope of the example and what is omitted.
+
+For example:
+
+> The following YAML example demonstrates the configuration of a single port numbered `80` using the `TCP` protocol.
+>
+> It is part of a Kubernetes Service specification.
+> It is not a complete Service specification and must be incorporated with the rest of a Service specification.
+>
+> ```yaml
+> ports:
+>   - port: 80
+>     protocol: TCP
+> ```
+
+#### Explain how to integrate the example
+
+With YAML examples especially, it is important to explain how to integrate the example within a wide configuration.
+
+Explain which key the example is the value for.
+The following example extends the one in the previous section:
+
+> The following YAML example demonstrates the configuration of a single port numbered `80` using the `TCP` protocol.
+>
+> It is part of a Kubernetes Service specification.
+> It is not a complete Service specification and must be incorporated with the rest of a Service specification.
+>
+> To incorporate the example, you must include it as the value to the Service `spec` mapping.
+> If there is an existing `ports` value, you must choose to replace it or merge the two.
```

Comment:
Question, do we want to mention providing the dot notation (not sure if that's what it's officially called) with the path to where the snippet should live in the file, for example "The following sample would be inserted into your Kubernetes Service file under `spec.ports`."

Reply:
I think we should for sure. I meant to add a comment to myself to that effect and clearly forgot :) 

I think it might require its own paragraph to explain the rules for the dot syntax.

Reply:
Yeah, it might need a paragraph of explanation at that.

>It turns out it's super hard!
Truth!